### PR TITLE
add ability to add contributor information to html pages (#1254)

### LINKED ISSUE #1254: Make list of maintainers for application definitions
One of the ideas that came out of the April 2023 telco ([minutes](https://www.nexusformat.org/Telco_20230419.html)), during discussion of #1011, was that it would be a good idea to add "maintainers" to application definitions.  These would be people who are most interested in guiding the maintenance and development of a given definition, or have an interest in it as a community member, or were long time contributors to it and can guide new users in its use or answer questions.  Critically, they wouldn't need to be NIAC members.

I'm happy to set this up, my question is, where should it live?

Proposals:

1. Add lists of application definitions to NIAC members on the [NIAC](https://www.nexusformat.org/NIAC.html#current-members) web page.
2. Add contact information (github handles?) to the [list of application definitions](https://manual.nexusformat.org/classes/applications/index.html)
3. Add contact information (github handles?) to individual definitions.  Perhaps in the the form of a new attribute or field?  Or in the doc string?

Or some combination of the above.  Thoughts on this?  Thanks!

### PR DESCRIPTION
**Pull request by RussBerg**
Ok I finally have all the code changes made to accomplish what was talked about, 
The contributors are listed left to right most recent to oldest, mouse over gives name and date of last commit and clicking avatar takes you to that persons github profile. 

Also added is the link to create a new github issue in the **get help** link.

Image

A github access token is required to retrieve the contributor information from the github API and this token is passed in the environment variable **GH_TOKEN**. During the codecamp @mkuehbach sent me an example of how they do it in fairmat https://github.com/FAIRmat-NFDI/nexus-fairmat-proposal/blob/main/.github/workflows/update-proposal-webpage.yml

So it looks like what we need can be achieved in jenkins by wrapping the line:

https://github.com/nexusformat/definitions/blob/2635a440a5d7cb0e521526a553116a9bc7aa0be0/jenkins_build#L30

like this?
```
withEnv(['GH_TOKEN=secrets.GITHUB_TOKEN']) {
         // Then call to build docs
         ( cd ../impatient-guide && rm -rf _build && make latexpdf && make html )
       
     }
```

please @prjemian could you check me on this
closes #1254

### CODE REVIEW

**Comment by prjemian** [comment]
@RussBerg Sorry, I won't have time to review in the next day or so.  Perhaps someone else can review?  I re-assigned to the NeXus developers

File: dev_tools/utils/github.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        if commit_dct["committer"] is not None:
+            contributor = commit_dct["commit"]["committer"]["name"]
+            if contributor in contrib_skip_list:
+                continue
+            contributor_names.add(contributor)
+            if commit_dct["commit"]["committer"]["email"] not in _email_lst:
+                _email = commit_dct["commit"]["committer"]["email"]
+                _email_lst.append(_email)
+                contribs_dct[commit_dct["commit"]['committer']["date"]] = {
+                    "name": format_author_name(commit_dct["commit"]["committer"]["name"]), "committer": commit_dct}
```
Comment (PeterC-DLS) [inline dev_tools/utils/github.py:80]: Pass back smaller dictionary with just items you need.

File: manual/source/_templates/sourcelink.html [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{#
+    basic/sourcelink.html
+    ~~~~~~~~~~~~~~~~~~~~~
+		'classes/applications/NXstxm'
```
Comment (PeterC-DLS) [inline manual/source/_templates/sourcelink.html:4]: ???

**Review by PeterC-DLS — COMMENTED** [review]
Run the `black` formatter

File: dev_tools/utils/github.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+import requests
+
+def format_author_name(nm):
+    """
+    make sure all words in name start with a capital
+    """
+    s = ""
+    nms = nm.split(" ")
+    for n in nms:
+        s += n[0].upper() + n[1:] + " "
```
Comment (PeterC-DLS) [inline dev_tools/utils/github.py:12]: Use string's join method

File: dev_tools/docs/nxdl.py [hunk 4]
```diff
@@ -100,6 +104,21 @@ def _parse_nxdl_file(self, nxdl_file: Path):
...
+        # add the contributors as variables to the rst file that will
+        nxdl_root = get_nxdl_root()
+        rel_path = str(nxdl_file.relative_to(nxdl_root))
+        rel_html = str(rel_path).replace(os.sep, "/")
+        contribs_dct = get_file_contributors_via_api("definitions", rel_html)
+        if contribs_dct is not None:
+            self._print("")
+            self._print("..")
+            self._print("    Contributors List")
+            i = 0
```
[the remark this answers is not in the document]
Comment (PeterC-DLS) [reply dev_tools/docs/nxdl.py:116]: I mean is `i` unused?

File: dev_tools/utils/github.py (hunk 1 above)
Comment (RussBerg) [reply dev_tools/utils/github.py:80]: my thought here was that down the road if we wanted to add for example an **Authors** section or something else all the information would already be there and would just need to be referenced during the rst line generation in nxdl.py,

File: manual/source/_templates/sourcelink.html (hunk 2 above)
Comment (RussBerg) [reply manual/source/_templates/sourcelink.html:4]: left over temporary reminder to myself

File: dev_tools/docs/nxdl.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (RussBerg) [reply dev_tools/docs/nxdl.py:116]: ahh yes, its been turfed 👍

**Comment by RussBerg** [comment]
@PeterC-DLS are you familiar with the jenkins stuff to hazard a guess on how to pull the github access token from "secrets" and add it to the environment?

**Comment by PeterC-DLS** [comment]
Yes, I can adjust the workflow to populate the environment running the build with the token. Note, the CI is failing as there are files not formatted correctly.

**Comment by PeterC-DLS** [comment]
See this change https://github.com/nexusformat/definitions/blob/76d84d6f56900b2a4c24020474f5c4e41d8f1025/.github/workflows/ci.yaml#L55-L56 and add it to https://github.com/nexusformat/definitions/blob/83498e7fbd02ed4eea664b9c37fef4e43a3b5498/.github/workflows/ci.yaml#L72

**Comment by PeterC-DLS** [comment]
Sorry, I clicked the wrong button!

File: dev_tools/utils/github.py (hunk 1 above)
Comment (PeterC-DLS) [reply dev_tools/utils/github.py]: Okay, that's fine. Perhaps change the key used to `"commit_dct"`.

Comment (RussBerg) [reply dev_tools/utils/github.py]: done!

**Comment by PeterC-DLS** [comment]
Does the CI GH token allow access to GH's REST API? If not, I could set an specific access token for this purpose

**Comment by RussBerg** [comment]
although I don't have any direct experience using the REST API it looks like (from what I read) we should be able to use the same token

**Review by PeterC-DLS — APPROVED** [review]
LGTM

**Pull request merged**